src/battle-scene.ts

```diff
@@ -2467,6 +2467,9 @@ export default class BattleScene extends SceneBase {
       const difficultyWaveIndex = this.gameMode.getWaveForDifficulty(this.currentBattle.waveIndex);
       const isFinalBoss = this.gameMode.isWaveFinal(this.currentBattle.waveIndex);
       let chances = Math.ceil(difficultyWaveIndex / 10);
+      if (this.getPlayerPokemon()?.getAbility().hasAttr(BonusItemChanceAbAttr) || this.getPlayerPokemon()?.getPassiveAbility().hasAttr(BonusItemChanceAbAttr)) {
```

Comment:
i think `hasAbility` should cover both active and passive abilities

Reply:
```suggestion
      if (this.getPlayerPokemon()?.hasAbilityWithAttr(BonusItemChanceAbAttr)) {
```
better to use `hasAbilityWithAttr` here